src/backend/uploader/imgur/ImgurWrapper.cpp

```diff
@@ -157,9 +157,9 @@ void ImgurWrapper::handleDataResponse(const QDomElement& element) const
         emit tokenRefreshRequired();
     } else {
         if (element.elementsByTagName(QLatin1String("error")).isEmpty()) {
-            emit error(QLatin1String("Server responded with ") + element.attribute(QLatin1String("status")));
+			emit error(QNetworkReply::ProtocolFailure, QLatin1String("Server responded with ") + element.attribute(QLatin1String("status")));
```

Comment:
Pretty sure ProtocolFailure is the best option for these errors.

Reply:
Sound good I think.

Reply:
Though I'm not sure if we should pass the QNetworkReply outside the Warpper. The idea of the wrapper is to hide the implementation, the upload could happen via network or be send via pigeon, the caller shouldn't know about it. Is there a way to not pass this enum outside the the Warpper? Haven't had time to look into it yet, I can check it over the weekend.

Reply:
Can't think of another way though :thinking:

Reply:
Looking at the code, is the error type that much relevant? In the path with the settings we ignore it completely. From my point of view the issue was `UploadStatus::NoError` in the `imgurError` method, there it should have been something like `UploadStatus::WebError`. How about moving the mapping method into the ImgurWrapper class and formatting the error string based on the type and only returning the string again. So or so we only forward this information to the notification service and show it to the user, we have no special treatment based on type. What do you think?